Sources/SwiftFormatPrettyPrint/PrettyPrint.swift

```diff
@@ -220,7 +220,11 @@ public class PrettyPrinter {
         lastBreakOffset = 0
         lastBreakValue = 0
       }
-      write(syntaxToken.text)
+      if syntaxToken.leadingTrivia.hasBackticks {
```

Comment:
The paranoid side of me wants to say that we should _technically_ look at the leading and trailing trivia separately, even if it's not something that is syntactically valid in the language today.

For example, I just ran ``let `x ` = 5`` through `-emit-syntax` and it does give us a syntax tree back—it ends up splitting the thing up into a `PatternBindingSyntax` with an `UnknownPattern` and an `UnknownStmtSyntax` for the rest of the line.

This actually brings up a separate issue—we should probably handle all `Unknown*Syntax` nodes by just writing out the tokens _verbatim_ so that we preserve what the user had instead of potentially making it worse. That would also address this specific situation; but I'm not sure what the right way to represent that in the pretty printer is. We could introduce yet another token case like `verbatim(TokenSyntax)` that literally does nothing more than append its text description, but introducing another case to handle feels a bit unfortunate...

What do you think?

Reply:
That is a good question, and I'm not exactly sure what to do about \`x \` or \`x. The latter case crashes the formatter with `unknownTokenKind("backtick")`. It may be something that is best handled in stage 1. Lone backticks are not allowed, are they?

For dealing with UnknownSyntax, creating a `verbatim` case might actually be the right way to go, since its printing and length calculation is going to be very different from other token types.

Reply:
If we go with the approach of "everything `Unknown*` is printed verbatim", then what you have here should be ok for the correct cases, where there will always be a backtick in the leading and trailing trivia so we shouldn't have to explicitly check it unless the language changes.

If that approach sounds good to you, go ahead and merge this. Thanks!